### PR TITLE
Migrate memory_instance_tree_view.dart to null safety.

### PR DESCRIPTION
**Pull request by polina-c**

### CODE REVIEW

File: packages/devtools_app/lib/src/screens/memory/memory_instance_tree_view.dart [hunk 1]
```diff
@@ -45,42 +44,43 @@ class InstanceTreeViewState extends State<InstanceTreeView>
...
+    return _controller.instanceFieldsTreeTable!;
   }
 }
 
 class _FieldTypeColumn extends TreeColumnData<FieldReference> {
   _FieldTypeColumn() : super('Type');
 
   @override
-  dynamic getValue(FieldReference dataObject) =>
+  Object? getValue(FieldReference dataObject) =>
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/memory/memory_instance_tree_view.dart:83]: non-nullable. Also should this be returning a String? it seems most other columns return a string value for their `getValue` method

File: packages/devtools_app/lib/src/screens/memory/memory_instance_tree_view.dart [hunk 2]
```diff
@@ -94,8 +94,8 @@ class _FieldTypeColumn extends TreeColumnData<FieldReference> {
 
   @override
   int compare(FieldReference a, FieldReference b) {
-    final Comparable valueA = getValue(a);
-    final Comparable valueB = getValue(b);
+    final Comparable valueA = getValue(a) as Comparable<Object>;
+    final Comparable valueB = getValue(b) as Comparable<Object>;
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/memory/memory_instance_tree_view.dart:98]: these casts should be able to be removed once getValue return type is changed to string. here and below.

File: packages/devtools_app/lib/src/screens/memory/memory_instance_tree_view.dart [hunk 3]
```diff
@@ -111,7 +111,7 @@ class _FieldNameColumn extends ColumnData<FieldReference> {
         );
 
   @override
-  dynamic getValue(FieldReference dataObject) =>
+  Object? getValue(FieldReference dataObject) =>
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/memory/memory_instance_tree_view.dart:114]: same comment: `String`, here and below.

File: packages/devtools_app/lib/src/screens/memory/memory_instance_tree_view.dart (hunk 1 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/memory/memory_instance_tree_view.dart]: done

File: packages/devtools_app/lib/src/screens/memory/memory_instance_tree_view.dart (hunk 3 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/memory/memory_instance_tree_view.dart]: done

File: packages/devtools_app/lib/src/screens/memory/memory_instance_tree_view.dart (hunk 2 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/memory/memory_instance_tree_view.dart]: done

**Review by kenzieschmoll — APPROVED** [review]

**Pull request merged**